credential/template/dataset.go

```diff
@@ -0,0 +1,127 @@
+package template
+
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	gotemplate "text/template"
+	"time"
+
+	"github.com/axone-protocol/axone-sdk/credential"
+	"github.com/axone-protocol/axone-sdk/dataverse"
+	"github.com/google/uuid"
+)
+
+//go:embed vc-desc-tpl.jsonld
+var datasetTemplate string
+
+var _ credential.Descriptor = NewDataset()
+
+// DatasetDescriptor is a descriptor for generate a dataset description VC.
+// See https://docs.axone.xyz/ontology/next/schemas/credential-dataset-description
+type DatasetDescriptor struct {
+	id           string
+	datasetDID   string
+	title        string
+	description  string
+	format       string
+	tags         []string
+	topic        string
+	issuanceDate *time.Time
+}
+
+func NewDataset() *DatasetDescriptor {
```

Comment:
```suggestion
func NewDataset(ID, title string, opts ...Option) *DatasetDescriptor {
```

By enforcing the presence of required fields and providing options for the rest we can avoid validating it at the generation step

Reply:
Yes, that seems better. It raises the question: which fields are required? Besides the `ID` and `issuanceDate`, which can be generated with default values (generated uuid fo `ID` and current date for `issuanceDate`), it appears that only the `title` is required ?

Reply:
Yes I think only those (`datasetDID & title`) are required, `issuanceDate` and `id` are mandatory too but default values can be generated as you did :)